pkg/steps/template.go

```diff
@@ -421,18 +421,50 @@ func createOrRestartPod(podClient coreclientset.PodInterface, pod *coreapi.Pod)
 	return created, nil
 }
 
-func waitForPodDeletion(podClient coreclientset.PodInterface, name string, uid types.UID) error {
+func waitForTeardownToComplete(pod *coreapi.Pod) {
 	timeout := 300
 	for i := 0; i < timeout; i += 2 {
-		pod, err := podClient.Get(name, meta.GetOptions{})
-		if errors.IsNotFound(err) {
-			return nil
-		}
-		if err != nil {
-			return fmt.Errorf("could not retrieve deleting pod: %v", err)
+		if !podTeardownCompleted(pod) {
```

Comment:
I pointed this out in the previous reviews, so I'll try to elaborate. I might be wrong, but I think we get `pod` as a param and it does not change at all while `waitForTeardownToComplete` executes (there is no API call that would update the pod status). So either container statuses in `pod` are from the start set that we would break on the first iteration, or they force us to wait, but `pod` won't change while we sleep, so we would always sleep the full 300s.

Reply:
ah, right, seems `waitForTeardownToComplete` should run `getPodByName` first, check that `teardown` container is present and then run the loop

Reply:
Reworked in c4d665d - now all functions are attempting to fetch pod info